src/core/util.cpp

```diff
@@ -211,6 +211,26 @@ namespace mamba
         return str.size() >= prefix.size() && 0 == str.compare(0, prefix.size(), prefix);
     }
 
+    bool any_starts_with(const std::vector<std::string_view>& str, const std::string_view& prefix)
+    {
+        for (auto& s : str)
+        {
+            if (starts_with(s, prefix))
+                return true;
+        }
+        return false;
+    }
+
+    bool starts_with_any(const std::string_view& str, const std::vector<std::string_view>& prefix)
```

Comment:
how about using std::any_of (https://en.cppreference.com/w/cpp/algorithm/all_any_none_of)

Reply:
Thanks for the comment I'll try to refactor that later